app/ui/legacy/src/main/java/com/fsck/k9/ui/K9Drawer.kt

```diff
@@ -282,6 +286,25 @@ class K9Drawer(private val parent: MessageList, savedInstanceState: Bundle?) : K
         }
     }
 
+    /**
+     * Custom layout for PrimaryDrawerItem allowing 2 lines
+     * and elipsize="middle" to keep the start+end of deep subfolder trees
+     * on screen.
+     *
+     * Can be replaced after
+     * https://github.com/mikepenz/MaterialDrawer/issues/2724
+     */
+    private class PrimaryFolderDrawerItem : PrimaryDrawerItem() {
```

Comment:
I tried that originally and somehow the layout was not picked up.
Since it's only a temporary solution until the elipsise can be set regularly, it was good enough.

Reply:
ktlintFormat doesn seem to work on Windows.
"Execution optimizations have been disabled for 68 invalid unit(s) of work during this build to ensure correctness."
no lint warning but after successfull execution it ends with a Exception because 
"Unable to watch the file system for changes"

So I can't see what it is that the CI pipeline here complained about.

Reply:
> I tried that originally and somehow the layout was not picked up.

You probably forgot to override the `type` property. Because the Unified Inbox is added first and is using `PrimaryDrawerItem` the adapter will always inflate the `PrimaryDrawerItem` layout for the shared type value.

See https://github.com/mikepenz/FastAdapter/blob/ac900c9a599b8ca6a1cf5ba2ef8022115fbc125b/fastadapter/src/main/java/com/mikepenz/fastadapter/FastAdapter.kt#L580-L587

> Since it's only a temporary solution until the elipsise can be set regularly, it was good enough.

You're assuming MaterialDrawer will be updated to support this use case. To me it feels like adding unnecessary API surface for something that won't be used by many people and can be accomplished using existing functionality. As a maintainer I'd probably reject the feature request.